eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -401,34 +398,93 @@ public void testInvalidConfigJSON() {
     }
 
     @Test
-    public void testCachedBadResponseAllowsLaterFetching() {
+    public void testCachedBadResponseRequiresFetch() {
         // Populate the cache with a bad response
-        ConfigCacheFile cacheFile = new ConfigCacheFile(ApplicationProvider.getApplicationContext());
-        cacheFile.delete();
-        try {
-            cacheFile.getOutputWriter().write("{}");
-            cacheFile.getOutputWriter().close();
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+        ConfigCacheFile cacheFile = new ConfigCacheFile(ApplicationProvider.getApplicationContext(), safeCacheKey(DUMMY_API_KEY));
+        cacheFile.setContents("{ invalid }");
```

Comment:
we want invalid JSON for this test as the empty object case is much more similar to the newly-added no flags test case